### DISCUSSION TITLE
IR remote issues

### DISCUSSION THREAD

**Discussion by ErnieBall** [rern/rAudio#978] [General]
i have followed this guide to get an IR remote to work with my PiFi Digi+ IR reciever.
https://vaughanharper.com/2020/08/12/configuring-an-infrared-remote-control-to-control-runeaudio-archlinux-without-needing-lirc/

I have gotten to the point where 
`ir-keytable -t` 
recognizes the button promts of my remote correctly (i think).
For example, pressing the "next track" button on my remote i get

1435.080082: lirc protocol(nec): scancode = 0xb2c1
1435.080128: event type EV_MSC(0x04): scancode = 0xb2c1
1435.080128: event type EV_KEY(0x01) key_down: KEY_RIGHT(0x006a)
1435.080128: event type EV_SYN(0x00).
1435.130094: lirc protocol(nec): scancode = 0xb2c1 repeat
1435.130124: event type EV_MSC(0x04): scancode = 0xb2c1
1435.130124: event type EV_SYN(0x00).
1435.270044: event type EV_KEY(0x01) key_up: KEY_RIGHT(0x006a)
1435.270044: event type EV_SYN(0x00).

From my understanding `KEY_RIGHT` should instruct MPD to skip to the next track by default.
However neither this button nor the ones mapped to KEY_LEFT and KEY_PLAY have any effect in rAudio1

I would appreciate it a lot, if someone could help me getting this to work,

**Comment by rern** [comment]
**rAudio** playback control responses to:
- `KEY_PLAY` 
- `KEY_PAUSE` 
- `KEY_STOP`
- `KEY_PREVIOUS`
- `KEY_NEXT`

A cheap USB PC Remote is the easiest to setup, just plug it in and start using. No configurations to set.
In fact, anything behaving like a keyboard with media keys will be fine. Buttons on  Bluetooth headsets control playback with these keys as well.

**Reply by ErnieBall** [reply]
Thanks for the reply!

I changed KEY_RIGHT to KEY_NEXT in my keymap file. It shows up correctly when testing inputs with `ir-keytable -t`
but it still doesnt skip tracks in rAudio1.

I also tested your other suggestion, since i had a USB IR Reciever, that acts as a keyboard. i verified, that it works on my computer,
but plugged into my Pi, it once again doesnt do anything inside rAudio. 
As a sanity check, i then used a regular wired keyboard with media keys and that also didnt work.

Am i missing something?

**Reply by rern** [reply]
Settings > Features > Browser on RPi - enable (Start X server for GUI and keyboard/mouse)

**Reply by ErnieBall** [reply]
That did the trick. I might have disabled that before, thinking it wasnt relevant form me.
Thanks for taking the time to help me out!
